src/Hamiltonians/GuidingVectorSampling.jl

```diff
@@ -0,0 +1,123 @@
+"""
+    GuidingVectorSampling
+
+Wrapper over any `AbstractHamiltonian` that implements guided vector a.k.a. guided wave
+function sampling. In this importance sampling scheme the Hamiltonian is modified as follows.
+
+```math
+\\tilde{H}_{ij} = v_i^{-1} H_{ij} v_j
+```
```

Comment:
So this looks wrong. It should be `v_i H_{ij} v_j^-1`.

Reply:
The idea is that you multiply the guiding vector to the coefficient vector. In the vector-matrix product of the transformed Hamiltonian with the transformed coefficient vector, the right index (j) is contracted with the index of the coefficient vector (c_j). In order to get a similarity transformation, the guiding vector needs to cancel out to become a factor of 1. Hence this specific order.